daffodil-io/src/main/scala/org/apache/daffodil/processors/charset/CharsetUtils.scala

```diff
@@ -69,6 +71,10 @@ object CharsetUtils {
   }
 
   val unicodeReplacementChar = '\uFFFD'
+
+  private def SDE(id: String, args: Any*): Nothing = {
+    throw new Exception(id.format(args: _*))
+  }
```

Comment:
We usually avoid throwing generic Exception's. It often makes it hard to know how to best handle the error. There's also no context associated with this SDE, like schema file or line number, which most SDE's have. I think that's because it just doesn't exist here. Which makes me think more that CharsetUtils should not be throwing any exceptions/SDE's but should instead rely on callers to handle that.